src/Symfony/Component/Messenger/EventListener/StopWorkerOnMemoryLimitListener.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\EventListener;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Messenger\Event\WorkerRunningEvent;
+
+/**
+ * @author Simon Delicata <simon.delicata@free.fr>
+ * @author Tobias Schultze <http://tobion.de>
+ */
+class StopWorkerOnMemoryLimitListener implements EventSubscriberInterface
+{
+    private $memoryLimit;
+    private $logger;
+    private $memoryResolver;
+
+    public function __construct(int $memoryLimit, LoggerInterface $logger = null, callable $memoryResolver = null)
```

Comment:
wouldn't it make sense to make the logger the third argument?

Reply:
The memory resolver is mainly for testing and it has been like this before.